scr/main/java/bittercode/constant/ResponseCode.java

```diff
@@ -0,0 +1,42 @@
+package com.bittercode.constant;
+
+import java.util.Arrays;
+import java.util.Optional;
+
+public enum ResponseCode {
+
+    SUCCESS(200, "OK"),
+    FAILURE(422, "Unprocessible Entity, Failed to Process"),
+    PAGE_NOT_FOUND(404, "The Page You are Searching For is Not available"),
+    ACCESS_DENIED(403, "Please Login First to continue"),
+    BAD_REQUEST(400, "Bad Request, Please Try Again"),
+    INTERNAL_SERVER_ERROR(500, "Internal Server Error, Try Again!!"),
+    DATABASE_CONNECTION_FAILURE(406,
+            "Unable to Connect to DB, Please Check your db credentials in application.properties"),
+    METHOD_NOT_ALLOWED(405, "Requested HTTP method is not supported by this URL"),
+
+    ;
+
+    private final String message;
+    private final int code;
+
+    ResponseCode(int code, String message) {
+        this.message = message;
+        this.code = code;
+    }
+
+    public String getMessage() {
+        return message;
+    }
+
+    public int getCode() {
+        return code;
+    }
+
+    // Reverse lookup methods
+    public static Optional<ResponseCode> getMessageByStatusCode(int statusCode) {
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/PrajaktaBendreBito/BookStore/pull/1#issuecomment-2438181340>#16fdcd</a><span> - 10/25/2024, 04:08 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The getMessageByStatusCode method uses Arrays.stream() to perform a linear search on the enum values, which can be inefficient for frequent lookups.
</div>
<div id="fix">
<b>Fix:</b> Replace the stream-based lookup with a static Map for constant-time lookups.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -6,6 +6,7 @@
  public enum ResponseCode {
 
      SUCCESS(200, "OK"),
 +    // ... other enum constants ...
      METHOD_NOT_ALLOWED(405, "Requested HTTP method is not supported by this URL"),
 
      ;
 @@ -15,6 +16,15 @@
      private final String message;
      private final int code;
 
 +    private static final Map<Integer, ResponseCode> CODE_MAP = new HashMap<>();
 +
 +    static {
 +        for (ResponseCode rc : values()) {
 +            CODE_MAP.put(rc.getCode(), rc);
 +        }
 +    }
 +
 +
      ResponseCode(int code, String message) {
          this.message = message;
          this.code = code;
 @@ -35,8 +45,7 @@
 
      // Reverse lookup methods
      public static Optional<ResponseCode> getMessageByStatusCode(int statusCode) {
 -        return Arrays.stream(ResponseCode.values())
 -                .filter(error -> error.getCode() == statusCode)
 -                .findFirst();
 +        return Optional.ofNullable(CODE_MAP.get(statusCode));
      }
  }
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged